tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -279,3 +280,166 @@ def test_45498(self):
+        """
+        Verify IAM user creation/deletion and get IAM user list while soft-failure in loop.
+        """
+        LOGGER.info("STARTED: Verify IAM user creation/deletion and get IAM user list while "
+                    "soft-failure in loop.")
+        LOGGER.info("Precondition: Create %s iam users for deletion", self.num_users)
+        del_users = self.mgnt_ops.create_account_users(nusers=self.num_users)
+
+        LOGGER.info("Step 1: Create multiple IAM users.")
+        new_users = self.mgnt_ops.create_account_users(nusers=10)
+        new_users_list = list(new_users.keys())
+        LOGGER.info("Step 1: Created multiple IAM users.")
+
+        LOGGER.info("Step 2: Start IAM user CRUD operations while performing soft failure on "
+                    "random control pods in loop. Verify existing IAM users persist.")
+        LOGGER.info("Scale replicas for control pod to %s", self.repl_num)
+        resp = self.node_master_list[0].create_pod_replicas(num_replica=self.repl_num,
+                                                            deploy=self.deply_name)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Get the list of control pods")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)
+        LOGGER.info("Total number of control pods in the cluster are: %s", self.repl_num)
+
+        iam_output = Queue()
+        event = threading.Event()
+        args = {'user_crud': True, 'num_users': self.num_users,
+                'output': iam_output, 'del_users_dict': del_users, 'header': self.header}
+        thr_iam = threading.Thread(target=self.ha_obj.iam_bucket_cruds,
+                                   args=(event,), kwargs=args)
+        thr_iam.daemon = True  # Daemonize thread
+        LOGGER.info("Starting background threads for IAM user creation/deletion.")
+        thr_iam.start()
+        for failure in range(HA_CFG["common_params"]["short_loop"]):
+            pod = secrets.SystemRandom().choice(pod_list)
+            LOGGER.info("Creating soft failure of %s for % loop", pod, failure)
+            self.node_master_list[0].kill_process_in_container(pod_name=pod,
+                                                               container_name=const.CORTX_CSM_POD,
+                                                               process_name=const.CSM_AGENT_PRC,
+                                                               safe_kill=True)
+            LOGGER.info("Step 2.1: Check cluster has failures.")
+            resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+            assert_utils.assert_false(resp[0], resp)
+            LOGGER.info("Step 2.1: Checked cluster has failures.")
+        LOGGER.info("Step 2: Started IAM user CRUD operations while performing soft failure on "
+                    "random control pods in loop. Verified existing IAM users persisted.")
+
+        time.sleep(self.csm_soft_delay)
```

Comment:
if we check this delay before and check that cluster is back online, I guess that would be better. Thoughts?

Reply:
> if we check this delay before and check that cluster is back online, I guess that would be better. Thoughts?

I was doing it after validating IAM crud operation. Taking in before.